app/src/main/java/com/example/userinfo/MainActivity.kt

```diff
@@ -0,0 +1,182 @@
+                        startActivity(intent)
+                    }
+                }
+            }
+            cancel.setOnClickListener() {
+                visibility=true
+                modifyVisibility(userName, email, number, pincode, address,validate,cancel, confirm, visibility)
+            }
+        }
+
+    }
+
+    fun validateAllFields(
+        userName: EditText,
+        email: EditText,
+        number: EditText,
+        pincode: EditText,
+        address: EditText
+    ): Boolean {
+
+        if (!isValidEmail(email.text.toString())) {
+            callToast( "Invalid Email Address")
+            return false
+        }
+        if (!isValidPinCode(pincode.text.toString())) {
+            callToast("Invalid Pincode")
+            return false
+        }
+        if (!isValidPhone(number.text.toString())) {
+            callToast("Invalid Phone Number")
+            return false
+        }
+        if(!isValidUserName(userName.text.toString())){
+            callToast("Invalid User Name")
+            return false
+        }
+        return true
+    }
+
+    private fun callToast(str:String){
+        Toast.makeText(applicationContext, str,
+            Toast.LENGTH_SHORT).show()
+    }
+
+    private fun checkAllFields(
+        userName: EditText,
+        email: EditText,
+        number: EditText,
+        pincode: EditText,
+        address: EditText
+    ): Boolean {
+
+        if (userName.text.trim().isEmpty()) {
+            Log.i("UserName", userName.id.toString())
+            callToast("Username shouldn't be empty")
+            return false
+        }
+        if (email.text.trim().isEmpty()) {
+            callToast("Email shouldn't empty")
+            return false
+        }
+        if(number.text.trim().isEmpty()) {
+            callToast("Phone number shouldn't be empty")
+            return false
+        }
+        if(pincode.text.trim().isEmpty()){
+            callToast("Pincode shouldn't be empty")
+            return false
+        }
+        if(address.text.trim().isEmpty()){
+            callToast("Address shouldn't be empty")
+            return false
+        }
+        return true
+    }
+    fun isValidEmail(str: String): Boolean {
+        return emailPattern.matcher(str).matches()
+    }
+
+    fun isValidPinCode(str: String): Boolean {
+        return pincodePattern.matcher(str).matches()
+    }
+
+    fun isValidPhone(str: String): Boolean {
+        return phoneNumberPattern.matcher(str).matches()
+    }
+
+    fun isValidUserName(str:String):Boolean{
+        return userNamePattern.matcher(str).matches()
+    }
+
+    fun modifyVisibility(
+        userName:EditText,
+        email: EditText,
+        number: EditText,
+        pincode: EditText,
+        address: EditText,
+        validate:Button,
+        cancel:Button,
+        confirm:Button,
+        visibility: Boolean
+    ){
+        userName.isEnabled=visibility
```

Comment:
We can use `Group` with ConstraintLayout, if we want to handle visibility of items of same usage
https://developer.android.com/reference/androidx/constraintlayout/widget/Group